docs/withdraw_manually.md

```diff
@@ -0,0 +1,108 @@
+# Withdraw Funds Manually
+
+This guide explains how to withdraw funds stored in:
+
+- **Agent Safe**
+- **Agent EOA**
+- **Master Safe** (includes agent bonds and security deposits)
+- **Master EOA**
+
+## Prerequisites
+
+- You have [Metamask](https://metamask.io/) installed in your browser and know how to execute transactions.
+- You understand Safe{Wallet}s and have some familiarity with the [Safe webapp](https://app.safe.global/).
+
+> [!IMPORTANT]
+> This guide does **not** cover removing external investments (e.g., liquidity pools) made by the agent. Ensure that the agent has withdrawn any such funds to the **Agent Safe** before proceeding.
+
+## Step 1: Withdraw from Agent EOA and Agent Safe
+
+To withdraw funds from the Agent EOA and Agent Safe you have to access contents stored in the `.operate` folder:
+
+- **Mac**: `/Users/YOUR_USERNAME/.operate`
+- **Windows**: `C:\Users\YOUR_USERNAME\.operate`
+- **Linux**: `/users/YOUR_USERNAME/`
+
+1. **Stop Pearl.**
+
+2. **Locate Agent Data.** Open the file `.operate/sc-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/config.json` in a text editor. If running multiple agents, ensure to select the correct `sc-xxxxxx...` folder. Search for:
+   1. `name`: Identifies the agent type (e.g., Trader, Optimus, ...).
+   2. `keys`: Contains the **Agent EOA address** and its **private key**.
+   3. `multisig`: Contains the **Agent Safe address**.
+   4. `token`: Contains the **Ownership NFT ID**.
+
+3. **Import the Agent EOA into Metamask.**
+   1. Copy the **Agent EOA private key**.
+   2. Open Metamask.
+   3. Go to *Add account or hardware wallet* &#8594; *Import account*.
+   4. Select *Private Key*, paste the key, and press *Import*.
+   5. Ensure the **Agent EOA address** is selected and connected to the correct chain.
+
+4. **Withdraw from Agent Safe**
+   1. Open the Safe webapp: https://app.safe.global/home?safe=gno:AGENT_SAFE_ADDRESS (replace `AGENT_SAFE_ADDRESS`).
+   2. Press *Connect* &#8594; *Metamask*.
+   3. Go to *Home* &#8594; *Send*.
+   4. Select the recipient, token, and amount. Follow instructions to **create** and **execute** the transaction.
+   5. Repeat for all tokens you wish to transfer.
+
+5. **Withdraw from Agent EOA.** Use Metamask to send funds/tokens to the recipient address.
+
+You must repeat this process for as many chains the agent is in.
+
+> [!IMPORTANT]
+> Withdraw from Agent Safe first because transactions require gas fees from the Agent EOA. If the Agent EOA is drained first, it will be unable to interact with the Agent Safe.
+
+## Step 2: Terminate On-Chain Service
+
+1. **Stop Pearl.**
+
+2. **Locate data in the Olas Registry page of the agent.**
+   1. Navigate to: https://registry.olas.network/gnosis/services/OWNERSHIP_NFT_ID (replace `OWNERSHIP_NFT_ID` from Step 1.2).
```

Comment:
also worth mentioning chain here (to replace `/gnosis` part)

Reply:
Actioned